mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -526,22 +529,26 @@ private boolean userHasLookupRightsOn(Mailbox mailbox, MailboxSession session) t
     public void deleteMailbox(final MailboxPath mailboxPath, final MailboxSession session) throws MailboxException {
         LOGGER.info("deleteMailbox {}", mailboxPath);
         assertIsOwner(session, mailboxPath);
-        final MailboxMapper mapper = mailboxSessionMapperFactory.getMailboxMapper(session);
+        MailboxMapper mapper = mailboxSessionMapperFactory.getMailboxMapper(session);
 
-        Mailbox mailbox = mapper.execute((Mapper.Transaction<Mailbox>) () -> {
-            final Mailbox mailbox1 = mapper.findMailboxByPath(mailboxPath);
+        mapper.execute((Mapper.Transaction<Mailbox>) () -> {
+            Mailbox mailbox1 = mapper.findMailboxByPath(mailboxPath);
             if (mailbox1 == null) {
                 throw new MailboxNotFoundException(mailboxPath);
             }
 
+            QuotaRoot quotaRoot = quotaRootResolver.getQuotaRoot(mailboxPath);
+            QuotaCount quotaCount = quotaManager.getMessageQuota(quotaRoot).getUsed();
+            QuotaSize quotaSize = quotaManager.getStorageQuota(quotaRoot).getUsed();
+
             // We need to create a copy of the mailbox as maybe we can not refer to the real
             // mailbox once we remove it
             SimpleMailbox m = new SimpleMailbox(mailbox1);
             mapper.delete(mailbox1);
+            dispatcher.mailboxDeleted(session, mailbox1, quotaRoot, quotaCount, quotaSize);
```

Comment:
why the event get the current quota? If the event is treated later that data won't be very accurate.

Reply:
are you sure getUsed() will return the quota only for this mailbox? Can you add an integration test to check this?
I mean 2 mailboxes, both with emails. You remove 1 mailbox, and the updated quota should match the remaining mailbox.